### ISSUE TITLE
Set GUI title to the current working directory

### ISSUE THREAD

**Issue by qwfy** [equalsraf/neovim-qt#294]
I often open several nvim-qt instances and often confused which is which, have to click every one of them to find out.

Will nvim-qt add a function/command that allows one to modify the window title?

**Comment by equalsraf** [comment]
isnt this somethign that can be done in vimscript already? A combination of `set titlestring` and dirchanged perhaps?

This https://github.com/neovim/neovim/issues/6260 seems similar to what you want but it uses the status line instead.

I did a quick test with this

```vim
augroup dirchange
    autocmd!
    autocmd DirChanged * let &titlestring=v:event['cwd']
augroup END
```

This needs further tweaking to check the event scope, and to set the correct title string on startup. Perhaps there is already a plugin that does all this?

**Comment by qwfy** [comment]
My bad, tried `titlestring` once before, didn't work, turns out it needs `title` to be on. Now it works.

**Comment by KevinJohnMulligan** [comment]
```vim
set title

augroup dirchange
    autocmd!
    autocmd DirChanged * let &titlestring=v:event['cwd']
augroup END
```

I can confirm that this sets the GUI window title to the file name and adds the the current working directory - Which is very useful! 

Thank you.

**Comment by rerrabolu** [comment]
Any ideas on getting this to work on nvim-qt.

**Comment by rerrabolu** [comment]
Hi Gehrig thanks for taking time to respond:

Just to add results of my experiment:

1. Make sure nvim config file "init.vim" **does not define** the parameter "title"
2. Define the parameter "title" for nvim-qt in its config file "ginit.vim"
3. e.g. set title

Once I did I was able to see nvim-qt window show filename in its title bar. Don't know how all of these things are inter-related. Happy to have it working.
